source/Files.cpp

```diff
@@ -338,67 +335,44 @@ string Files::Name(const filesystem::path &path)
 
 
 
-FILE *Files::Open(const filesystem::path &path, bool write)
+shared_ptr<iostream> Files::Open(const filesystem::path &path, bool write)
 {
-#if defined _WIN32
-	FILE *file = nullptr;
-	_wfopen_s(&file, path.c_str(), write ? L"w" : L"rb");
-	return file;
-#else
-	return fopen(path.c_str(), write ? "wb" : "rb");
-#endif
+	if(write)
+		return shared_ptr<iostream>{new fstream(path, fstream::out | fstream::binary)};
+	return shared_ptr<iostream>{new fstream(path, fstream::in | fstream::binary)};
```

Comment:
If we're using curly braces, we should use them consistently. Also I'd suggest `ios` as our style choice (here's `fstream` and in another file in this PR there's `iostream`).
```suggestion
		return shared_ptr<iostream>{new fstream{path, ios::out | ios::binary}};
	return shared_ptr<iostream>{new fstream{path, ios::in | ios::binary}};
```